x-pack/functionbeat/provider/aws/sqs.go

```diff
@@ -101,12 +102,51 @@ func (s *SQS) Template() *cloudformation.Template {
 	return template
 }
 
+// Policies returns a slice of policies to add to the lambda role.
+func (s *SQS) Policies() []cloudformation.AWSIAMRole_Policy {
+	resources := make([]string, len(s.config.Triggers))
+	for idx, trigger := range s.config.Triggers {
+		resources[idx] = trigger.EventSourceArn
+	}
+
+	// Give us a chance to generate the same document indenpendant of the changes,
+	// to help with updates.
+	sort.Strings(resources)
+
+	// SQS Roles permissions:
+	// - lambda:CreateEventSourceMapping
+	// - lambda:ListEventSourceMappings
+	// - lambda:ListFunctions
+	//
+	// Lambda Role permission
+	// - sqs:ChangeMessageVisibility
+	// - sqs:DeleteMessage
+	// - sqs:GetQueueAttributes
+	// - sqs:ReceiveMessage
+	policies := []cloudformation.AWSIAMRole_Policy{
+		cloudformation.AWSIAMRole_Policy{
+			PolicyName: cloudformation.Join("-", []string{"fnb", "sqs", s.config.Name}),
+			PolicyDocument: map[string]interface{}{
+				"Statement": []map[string]interface{}{
+					map[string]interface{}{
+						"Action": []string{
+							"sqs:ChangeMessageVisibility",
+							"sqs:DeleteMessage",
```

Comment:
Just curious: Why is this permission needed?

Reply:
It's linked to how SQS works, when you receive a batch of events from SQS you have to change the visibility of the events from _visible_ to _hidden_, that visibility property has a timeout, this allows an event to be used by another workers if your initial worker crashed or something else bad happened.

When you have successfully did the work on the events you have to delete the event from the queue.

Now looking at the official [doc](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html)
They list the following:

    - sqs:ReceiveMessage
    - sqs:DeleteMessage
    - sqs:GetQueueAttributes


And another official doc (I can't find it) also mention `sqs:ChangeMessageVisibility`.

Reply:
Thank you for the explanation.